cpmpy/solvers/minizinc.py

```diff
@@ -110,22 +112,68 @@ def __init__(self, cpm_model=None, solver=None):
             solvername = solvername[9:]
         self.name = "minizinc:"+solvername
 
-        import minizinc
+        # Create solver specific objects
         self.mzn_solver = minizinc.Solver.lookup(solvername)
-
-        # minizinc-python API
-        # Create a MiniZinc model
         self.mzn_model = minizinc.Model()
-        if cpm_model is None:
-            self.user_vars = []
-            self.mzn_txt_solve = "solve satisfy;"
-        else:
-            # store original vars and objective (before flattening)
-            self.user_vars = get_variables_model(cpm_model)
-            (mzn_txt, self.mzn_txt_solve) = self.make_model(cpm_model)
-            self.mzn_model.add_string(mzn_txt)
-            # do NOT add self.mzn_txt_solve yet, so that it can be overwritten later
+        self.mzn_model.add_string(CPM_minizinc.header)
+
+        #Solver text
+        self.mzn_txt_solve = "solve satisfy;"
+
+        super().__init__(cpm_model, solver=solver, name=solvername)
+
+    def __add__(self, cons):
+        """
+            Add an additional (list of) constraints to the model
+        """
+        # Add variables in cons
+        for var in get_variables(cons):
+            if var in self.user_vars:  # Might be possible to do this prettier?
+                continue
+            if isinstance(var, _BoolVarImpl):
```

Comment:
this should go in solver_var() as for the other interfaces, see another comment below